_pytest/recwarn.py

```diff
@@ -28,9 +28,18 @@ def pytest_namespace():
             'warns': warns}
 
 
-def deprecated_call(func, *args, **kwargs):
+def deprecated_call(func=None, *args, **kwargs):
     """Assert that ``func(*args, **kwargs)`` triggers a DeprecationWarning.
+
+    This function can be used as a context manager::
+
+        >>> with deprecated_call():
+        ...    myobject.deprecated_method()
     """
+    if not func:
+        warnings.simplefilter('always')
```

Comment:
append decides whether the filter is inserted at the end or the beginning in the warnings filter list